test/unit/org/apache/cassandra/audit/AuditLoggerAuthTest.java

```diff
@@ -274,19 +275,19 @@ private static void assertSource(AuditLogEntry logEntry, String username)
             assertEquals(username, logEntry.getUser());
     }
 
-    private static String getCreateRoleCql(String role, boolean login, boolean superUser)
+    private static String getCreateRoleCql(String role, boolean login, boolean superUser, boolean isPasswordObfuscated)
     {
-        return String.format("CREATE ROLE IF NOT EXISTS %s WITH LOGIN = %s AND SUPERUSER = %s AND PASSWORD = '%s'",
-                             role, login, superUser, TEST_PW);
+        String baseQueryString = String.format("CREATE ROLE IF NOT EXISTS %s WITH LOGIN = %s AND SUPERUSER = %s AND PASSWORD", role, login, superUser);
+        return isPasswordObfuscated ? baseQueryString + PasswordObfuscator.OBFUSCATION_TOKEN : baseQueryString + String.format(" = '%s'", TEST_PW);
     }
 
     private static void createTestRole()
     {
-        String createTestRoleCQL = getCreateRoleCql(TEST_ROLE, true, false);
+        String createTestRoleCQL = getCreateRoleCql(TEST_ROLE, true, false, false);
         executeWithCredentials(Arrays.asList(createTestRoleCQL), CASS_USER, CASS_PW, AuditLogEntryType.LOGIN_SUCCESS);
         assertTrue(getInMemAuditLogger().size() > 0);
         AuditLogEntry logEntry = getInMemAuditLogger().poll();
-        assertLogEntry(logEntry, AuditLogEntryType.CREATE_ROLE, createTestRoleCQL, CASS_USER);
+        assertLogEntry(logEntry, AuditLogEntryType.CREATE_ROLE, getCreateRoleCql(TEST_ROLE, true, false, true), CASS_USER);
         assertEquals(0, getInMemAuditLogger().size());
     }
 }
```

Comment:
Is it possible to add a test case covering an invalid DCL statement, something that you identified in review? if it is already added, ignore my comment.

`INFO  [Native-Transport-Requests-1] 2021-06-09 23:38:05,965 FileAuditLogger.java:51 - user:anonymous|host:localhost/127.0.0.1:7000|source:/127.0.0.1|port:63360|timestamp:1623296285965|type:REQUEST_FAILURE|category:ERROR|operation:BEGIN BATCH CREATE ROLE alice WITH PASSWORD = 'password_a' AND LOGIN = true; CREATE ROLE new_role; APPLY BATCH;; line 1:12 mismatched input 'CREATE' expecting K_APPLY (BEGIN BATCH [CREATE]...)
`

**a rough test case that could cover this case** 

```

public void testCqlCreateRoleSyntaxError()
{ 
          String createTestRoleCQL = String.format("CREATE ROLE %s WITH LOGIN = %s ANDSUPERUSER = %s AND PASSWORD", TEST_ROLE, true, false) +PasswordObfuscator.OBFUSCATION_TOKEN;

        executeWithCredentials(Arrays.asList(createTestRoleCQL), CASS_USER, CASS_PW, AuditLogEntryType.LOGIN_SUCCESS);
        assertTrue(getInMemAuditLogger().size() > 0);
        AuditLogEntry logEntry = getInMemAuditLogger().poll();
        assertLogEntry(logEntry, AuditLogEntryType.REQUEST_FAILURE, ...
```